src/components/Router/index.js

```diff
@@ -42,10 +42,18 @@ class CrossDeviceMobileRouter extends Component {
 
   setConfig = (actions) => (data) => {
     const {token, steps, documentType, step} = data
-    this.setState({token, steps, step})
+    const crossDeviceClientSteps = this.replaceLastStep(steps)
+    this.setState({token, steps: crossDeviceClientSteps, step})
     actions.setDocumentType(documentType)
   }
 
+  replaceLastStep = (steps) => {
+    if (steps[steps.length - 1].type === 'complete') {
+      steps.pop()
```

Comment:
`steps` is state in this case, but in reality is a props that came from the cross device master, as such it should be used as a means to generate new data (eg. the componentsList), so we should avoid changing it directly.